ibc/docs/ics02.md

```diff
@@ -0,0 +1,156 @@
+
+## Implementation
+
+[ICS-02][ics02] is implemented according to the specification. However as the Pocket protocol will utilise [ICS-08][ics08] WASM clients for the improvements to client upgradeability; the implementations of the `ClientState`, `ConsensusState` and other interfaces are specific to a WASM client.
+
+The implementation details are explored below, the code for ICS-02 can be found in [ibc/client](../client/)
+
+### Client Manager
+
+The `ClientManager` is the submodule that governs the light client implementations and implements the [ICS-02][ics02] interface. It is defined in [shared/modules/ibc_client_module.go](../../shared/modules/ibc_client_module.go). The `ClientManager` exposed the following methods:
+
+```go
+// === Client Lifecycle Management ===
+
+// CreateClient creates a new client with the given client state and initial consensus state
+// and initialises its unique identifier in the IBC store
+CreateClient(ClientState, ConsensusState) (string, error)
+
+// UpdateClient updates an existing client with the given ClientMessage, given that
+// the ClientMessage can be verified using the existing ClientState and ConsensusState
+UpdateClient(identifier string, clientMessage ClientMessage) error
+
+// UpgradeClient upgrades an existing client with the given identifier using the
+// ClientState and ConsenusState provided. It can only do so if the new client
+// was committed to by the old client at the specified upgrade height
+UpgradeClient(
+	identifier string,
+	clientState ClientState, consensusState ConsensusState,
+	proofUpgradeClient, proofUpgradeConsState []byte,
+) error
+
+// === Client Queries ===
+
+// GetConsensusState returns the ConsensusState at the given height for the given client
+GetConsensusState(identifier string, height Height) (ConsensusState, error)
+
+// GetClientState returns the ClientState for the given client
+GetClientState(identifier string) (ClientState, error)
+
+// GetHostConsensusState returns the ConsensusState at the given height for the host chain
+GetHostConsensusState(height Height) (ConsensusState, error)
+
+// GetHostClientState returns the ClientState at the provided height for the host chain
+GetHostClientState(height Height) (ClientState, error)
+
+// GetCurrentHeight returns the current IBC client height of the network
+GetCurrentHeight() Height
+
+// VerifyHostClientState verifies the client state for a client running on a
+// counterparty chain is valid, checking against the current host client state
+VerifyHostClientState(ClientState) error
+```
+
+#### Lifecycle Management
```

Comment:
Per my comment above, I think linking to the code and having a diagram with a lifecycle here will go a long way.